tests/infrastructure/tekton/conftest.py

```diff
@@ -323,38 +321,23 @@ def quay_disk_uploader_secret(custom_pipeline_namespace):
         yield quay_disk_uploader_secret
 
 
-@pytest.fixture()
-def created_fedora_dv_for_disk_uploader(admin_client, golden_images_namespace, custom_pipeline_namespace):
-    with create_dv(
-        dv_name="fedora-dv-disk-uploader",
-        namespace=custom_pipeline_namespace.name,
-        source=PVC,
-        source_pvc=DataSource(
-            name=OS_FLAVOR_FEDORA, namespace=golden_images_namespace.name
-        ).instance.spec.source.pvc.name,
-        source_namespace=golden_images_namespace.name,
-        size="35Gi",  # Keeping size of the data volume, slightly larger than the data source (~32Gi)
-        client=admin_client,
-        storage_class=py_config["default_storage_class"],
-    ) as dv:
-        dv.wait_for_dv_success()
-        yield dv
-
-
-@pytest.fixture()
-def vm_for_disk_uploader(custom_pipeline_namespace, admin_client, created_fedora_dv_for_disk_uploader):
+@pytest.fixture(scope="module")
+def vm_for_disk_uploader(custom_pipeline_namespace, admin_client, golden_images_namespace):
     with VirtualMachineForTests(
         name="fedora-vm-diskuploader",
         namespace=custom_pipeline_namespace.name,
         client=admin_client,
-        data_volume=created_fedora_dv_for_disk_uploader,
-        vm_instance_type=VirtualMachineClusterInstancetype(name="u1.small"),
-        vm_preference=VirtualMachineClusterPreference(name=OS_FLAVOR_FEDORA),
+        data_volume_template=data_volume_template_with_source_ref_dict(
+            data_source=DataSource(name=OS_FLAVOR_FEDORA, namespace=golden_images_namespace.name),
+            storage_class=py_config["default_storage_class"],
+        ),
+        vm_instance_type_infer=True,
+        vm_preference_infer=True,
```

Comment:
i think the param names are not really clear..
how about `vm_instance_type/preference_fetch`?

Reply:
I am using vm.spec.preference.inferFromVolume & vm.spec.instancetype.inferFromVolume
DESCRIPTION:
    InferFromVolume lists the name of a volume that should be used to infer or
    discover the instancetype
    to be used through known annotations on the underlying resource. Once
    applied to the InstancetypeMatcher
    this field is removed.

Reply:
it should be str and not bool. and the name of volume should be passed there (and not hard-codded set based on bool value)

Reply:
when vm_preference_infer or vm_instance_type_infer is set, it will pick dv_disk and link it to fedora datavolume.
there are defined as bool, https://github.com/RedHatQE/openshift-virtualization-tests/blob/main/utilities/virt.py#L346

Reply:
i think ideally we should change wrapper and generate in with class_generator.But too many api calls needed to be fixed

Reply:
agreed. can be changed in separate PR